sklearn/utils/__init__.py

```diff
@@ -18,7 +18,7 @@
 from .class_weight import compute_class_weight, compute_sample_weight
 from ._joblib import cpu_count, Parallel, Memory, delayed, hash
 from ._joblib import parallel_backend, register_parallel_backend
-from ._joblib import effective_n_jobs
+from ._joblib import effective_n_jobs, joblib
```

Comment:
> Note that for testing purposes, it is necessary to add the module joblib in utils to be able to access more advanced functionality of joblib without importing directly the module. 

It's only necessary for `test_forest`? Maybe keep it as a private object then e.g. `_joblib_module` or something?

Reply:
Would you agree to keeping `joblib` in `sklearn.utils._joblib`? this way it is clear that it is private, but We can still access it when needed.

Reply:
Finally, the package is used in multiple place so I just fixed the `flake8` error and replace all mention to `external.joblib` by `utils.joblib`.